### PR TITLE
Fix broken mac tests

### PR DESCRIPTION
**Pull request by aishfenton**
* Add explicit shutdown call to `LSPTestProcessor`. This should fix flaky macos tests.

### CODE REVIEW

File: src/main/scala/afenton/bazel/bsp/BazelBspServer.scala [hunk 1]
```diff
@@ -245,7 +245,7 @@ class BazelBspServer(
   def buildShutdown(params: Unit): IO[Unit] =
     for
       _ <- logger.info("build/shutdown")
-      _ <- IO.raiseError(new Exception("SHUTTING DOWN"))
+      // _ <- IO.raiseError(new Exception("SHUTTING DOWN"))
```
Comment (johnynek) [inline src/main/scala/afenton/bazel/bsp/BazelBspServer.scala:248]: can we remove the comment?

Why do we need this method? It looks like just a log call.

File: src/test/scala/afenton/bazel/bsp/LspTestProcess.scala [hunk 2]
```diff
@@ -87,8 +87,11 @@ case class BspClient(
   def buildInitialized(params: Unit): IO[DeferredSource[IO, Unit]] =
     sendRequest("build/initialized", params)
 
-  // def buildShutdown(params: Unit): IO[Unit]
-  // def buildExit(params: Unit): IO[Unit]
+  def buildShutdown(params: Unit): IO[DeferredSource[IO, Unit]] =
+    sendRequest("build/shutdown", params)
+
+  // def buildExit(params: Unit): IO[Unit] =
```
Comment (johnynek) [inline src/test/scala/afenton/bazel/bsp/LspTestProcess.scala:93]: can we remove comment

File: src/test/scala/afenton/bazel/bsp/LspTestProcess.scala [hunk 3]
```diff
@@ -204,13 +204,17 @@ case class LspTestProcess(workspaceRoot: Path):
...
-        ) <& processBspErr(Stream.fromQueueUnterminated(bspErrQ), duration))
+        processActions(client, actions, exitSwitch),
+        IO.race(
+          processBspOut(
+            Stream.fromQueueUnterminated(bspOutQ),
+            openRequests,
+            exitSwitch
+          ),
+          processBspErr(Stream.fromQueueUnterminated(bspErrQ))
+        ).map(_.left.get)
```
Comment (johnynek) [inline src/test/scala/afenton/bazel/bsp/LspTestProcess.scala:217]: I don't love the `left.get` but honestly at first glance I can't see all what is happening here.

File: src/test/scala/afenton/bazel/bsp/LspTestProcess.scala [hunk 4]
```diff
@@ -232,8 +236,22 @@ case class LspTestProcess(workspaceRoot: Path):
       _ <- d2.get
     yield resp
 
+  private def shutdown(
+      client: BspClient,
+      exitSwitch: Deferred[IO, Either[Throwable, Unit]]
+  ): IO[Unit] =
+    for
+      d1 <- client.buildShutdown(())
+      resp <- d1.get
```
Comment (johnynek) [inline src/test/scala/afenton/bazel/bsp/LspTestProcess.scala:245]: `resp` is never used. I think `_ <- d1.get` is what you want, you are just blocking until the d1 is completed, right?

File: src/test/scala/afenton/bazel/bsp/LspTestProcess.scala [hunk 5]
```diff
@@ -232,8 +236,22 @@ case class LspTestProcess(workspaceRoot: Path):
...
+      // d2 <- client.buildExit(())
+      // _ <- d2.get
+      _ <- exitSwitch.complete(Right(()))
+    yield ()
+
   extension [A](io: IO[A])
-    def marker(str: String) = io.map { a => System.err.println(str + s" WITH: $a");  a }
+    def marker(str: String) = io.map { a =>
+      System.err.println(str + s" WITH: $a"); a
```
Comment (johnynek) [inline src/test/scala/afenton/bazel/bsp/LspTestProcess.scala:253]: nit:
```scala
def marker(str: String): IO[A] =
  (for {
    either <- io.attempt
    _ <- IO.blocking(System.err.println(s"$str WITH: $either"))
  } yield a).rethrow
```

File: src/main/scala/afenton/bazel/bsp/BazelBspServer.scala (hunk 1 above)
Comment (aishfenton) [reply src/main/scala/afenton/bazel/bsp/BazelBspServer.scala:248]: Needs to shutdown server, changed in https://github.com/aishfenton/bazel-bsp/pull/6

File: src/test/scala/afenton/bazel/bsp/LspTestProcess.scala (hunk 2 above)
Comment (aishfenton) [reply src/test/scala/afenton/bazel/bsp/LspTestProcess.scala:93]: Added TODO for it. We should be sending it, but first have to support sending `Notification`'s vs `Request`'s

File: src/test/scala/afenton/bazel/bsp/LspTestProcess.scala (hunk 3 above)
Comment (aishfenton) [reply src/test/scala/afenton/bazel/bsp/LspTestProcess.scala:217]: Yeah, need some help simplifying all this. Check your slack :)

**Review by johnynek — APPROVED** [review]
we can improve the hairy bit later.

**Pull request merged**